zone/zone.cpp

```diff
@@ -2583,27 +2584,29 @@ void Zone::DoAdventureActions()
 
 }
 
-void Zone::LoadNPCEmotes(std::vector<NPC_Emote_Struct*>* NPCEmoteList)
+void Zone::LoadNPCEmotes(std::vector<NPC_Emote_Struct*>* v)
 {
+	v->clear();
 
-	NPCEmoteList->clear();
-    const std::string query = "SELECT emoteid, event_, type, text FROM npc_emotes";
-    auto results = content_db.QueryDatabase(query);
-    if (!results.Success()) {
-        return;
-    }
+	const auto& l = NpcEmotesRepository::All(content_db);
 
-    for (auto row = results.begin(); row != results.end(); ++row)
-    {
-	    auto nes = new NPC_Emote_Struct;
-	    nes->emoteid = Strings::ToInt(row[0]);
-	    nes->event_ = Strings::ToInt(row[1]);
-	    nes->type = Strings::ToInt(row[2]);
-	    strn0cpy(nes->text, row[3], sizeof(nes->text));
-	    NPCEmoteList->push_back(nes);
-    }
+	for (const auto& e : l) {
+		auto n = new NPC_Emote_Struct;
```

Comment:
Do we really need to allocate `new` for this ?

Reply:
Not allocating new it seems to break all emotes.